### PR TITLE
Update logic and labels for settings updates preferences in Cloud and Enterprise

### LINKED ISSUE #16792: Do not show Cloud CTA for Enterprise Edition
**Describe the bug**
Continuation of #11336. Regression since 1.39.0
Do not show CTA on Enterprise Edition. And do not show update panel at all on Metabase Cloud.

**To Reproduce**
1. Admin > Settings > Updates
image

**Expected behavior**
- [ ] Do not show CTA on Enterprise Edition
- [ ] Do not update panel at all on Metabase Cloud. Show a custom message.

**Information about your Metabase Installation:**
Regression since 1.39.0 (#15517)

### PR DESCRIPTION
**Pull request by gusaiani**
Fixes #16792

Intend to follow-up with e2e test PR.

### Cloud installations

In https://github.com/metabase/metabase/blob/master/frontend/src/metabase/lib/settings.js#L91, introduce an early return forcing `true`:

~~~js
  isHosted() {
    /* eslint-disable */
    return true
    
    // matches <custom>.metabaseapp.com and <custom>.metabaseapp.com/
    return /.+\.metabaseapp.com\/?$/i.test(this.get("site-url"));
  }
~~~

Visit http://localhost:3000/admin/settings/updates

You should:

1. no longer see a toggle to enable/disable updates
2. no longer see a number 1 beside "Updates" in the left menu, as a counter of available updates.
2. see the message `Metabase Cloud keeps your instance up to date. Thanks for being a customer!`

<hr />

### Enterprise 

Start application in [Enterprise mode](https://github.com/metabase/metabase/blob/master/enterprise/README.md)

Manually populate `version-info` and `version-info-last-checked` in the `setting` table so that it triggers an update check (I don't know how to do this atm, please show me how).

You should no longer see a call to action about upgrading to Enterprise.

### CODE REVIEW

File: frontend/src/metabase/admin/settings/components/SettingsUpdatesForm.jsx
Comment (ariya) [inline frontend/src/metabase/admin/settings/components/SettingsUpdatesForm.jsx:84]: JFYI, we shouldn't introduce a new string (for i18n) in the last minute before v40 release.

Comment (gusaiani) [reply frontend/src/metabase/admin/settings/components/SettingsUpdatesForm.jsx:84]: @ariya Yes, I was going to confirm this for this specific situation. Thanks!

Comment (gusaiani) [reply frontend/src/metabase/admin/settings/components/SettingsUpdatesForm.jsx:84]: Increasingly thinking that getting to the bottom of this i18n flow and being able to quickly turn around translations _might_ make us way faster and less prone to dropping balls in the 3-month-Metabase-zen-time-horizon.

Aware of the large queue of priorities we have, not to worry.

Comment (gusaiani) [reply frontend/src/metabase/admin/settings/components/SettingsUpdatesForm.jsx:86]: Separate ticket for this, to be done after v40.

https://github.com/metabase/metabase/issues/16923

Queued in board https://github.com/orgs/metabase/projects/16

File: frontend/src/metabase/admin/settings/components/SettingsUpdatesForm.jsx
Comment (gusaiani) [inline frontend/src/metabase/admin/settings/components/SettingsUpdatesForm.jsx:24]: Aware that introducing a new string atm is not ideal.

Please see discussion: https://github.com/metabase/metabase/issues/16792#issuecomment-875655045

cc @mazameli

Comment (mazameli) [reply frontend/src/metabase/admin/settings/components/SettingsUpdatesForm.jsx:24]: It's not a huge deal. Just let me know as soon as we merge this and I can get it translated. If we're going to do this one, incidentally, we may as well merge in that other typo PR you pinged me about, too.

File: frontend/src/metabase/admin/settings/components/SettingsUpdatesForm.jsx
Comment (gusaiani) [reply frontend/src/metabase/admin/settings/components/SettingsUpdatesForm.jsx:84]: Updated again, Maz is 👍 for this situation: https://github.com/metabase/metabase/pull/16920#discussion_r665731675

File: frontend/src/metabase/admin/settings/components/SettingsUpdatesForm.jsx
Comment (alxnddr) [inline frontend/src/metabase/admin/settings/components/SettingsUpdatesForm.jsx:141]: It looks like there are some redundant checks left

With early return when the instance is hosted, we can just add
```
// line: 29
const shouldShowHostedCta = !MetabaseSettings.isEnterprise()

// lines 36 and 80
{shouldShowHostedCta && <HostingCTA />}
```
What do you think?

File: frontend/src/metabase/admin/settings/containers/SettingsEditorApp.jsx
Comment (alxnddr) [inline frontend/src/metabase/admin/settings/containers/SettingsEditorApp.jsx:197]: optional nitpick:

Remove this `if` statement and change the jsx below

```
          {shouldDisplayNewVersionIndicator && (
            <span
              style={{ padding: "4px 8px 4px 8px" }}
              className="bg-brand rounded text-white text-bold h6"
            >
              1
            </span>
          )}
```

**Review by alxnddr — APPROVED** [review]
Overall looks good! I'd wait for @flamber approval

File: frontend/src/metabase/admin/settings/containers/SettingsEditorApp.jsx
Comment (gusaiani) [reply frontend/src/metabase/admin/settings/containers/SettingsEditorApp.jsx]: Done.

Yeah, looking at this file as a whole, the impulse is to refactor a lot more.

Thanks for the good nudge!

File: frontend/src/metabase/admin/settings/components/SettingsUpdatesForm.jsx
Comment (gusaiani) [reply frontend/src/metabase/admin/settings/components/SettingsUpdatesForm.jsx:140]: @alxnddr Yes, totally. Changed. Thank you very much for pointing it out. 🙇

**Pull request merged**
